### PR TITLE
Plotting Int64 columns with nulled integers (NAType) fails #32073

### LINKED ISSUE #32073: Plotting Int64 columns with nulled integers (NAType) fails
#### Code Sample, a copy-pastable example if possible

```
import pandas as pd
import numpy as np

df = pd.DataFrame({'A': [1, 2, 3, 4, 5], 'B': [7, 5, np.nan, 3, 2]})
df.plot(x='A', y='B')
df = df.astype('Int64')
df.plot(x='A', y='B')
```
#### Problem description

The first plotting command works, the second throws the error message

> TypeError: float() argument must be a string or a number, not 'NAType'

#### Expected Output

NAType should be treated the same way as numpy nan in plotting. Maybe transformed on the fly?

(I'm unsure if this is a pandas, a numpy, or a matplotlib issue, I'm starting here)

#### Output of ``pd.show_versions()``

<details>

INSTALLED VERSIONS
------------------
commit           : None
python           : 3.7.6.final.0
python-bits      : 64
OS               : Windows
OS-release       : 10
machine          : AMD64
processor        : Intel64 Family 6 Model 79 Stepping 1, GenuineIntel
byteorder        : little
LC_ALL           : None
LANG             : en
LOCALE           : None.None

pandas           : 1.0.1
numpy            : 1.18.1
pytz             : 2019.3
dateutil         : 2.8.1
pip              : 20.0.2
setuptools       : 45.2.0.post20200209
Cython           : 0.29.15
pytest           : None
hypothesis       : None
sphinx           : 2.4.1
blosc            : None
feather          : None
xlsxwriter       : 1.2.7
lxml.etree       : 4.5.0
html5lib         : 1.0.1
pymysql          : None
psycopg2         : None
jinja2           : 2.11.1
IPython          : 7.12.0
pandas_datareader: None
bs4              : 4.8.2
bottleneck       : None
fastparquet      : 0.3.3
gcsfs            : None
lxml.etree       : 4.5.0
matplotlib       : 3.1.3
numexpr          : None
odfpy            : None
openpyxl         : 3.0.3
pandas_gbq       : None
pyarrow          : 0.16.0
pytables         : None
pytest           : None
pyxlsb           : None
s3fs             : None
scipy            : 1.4.1
sqlalchemy       : 1.3.13
tables           : None
tabulate         : None
xarray           : None
xlrd             : 1.2.0
xlwt             : None
xlsxwriter       : 1.2.7
numba            : 0.48.0

</details>

### PR DESCRIPTION
**Pull request by jeandersonbc**
- [x] closes #32073
- [ ] tests added / passed
- [x] passes `black pandas`
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [ ] whatsnew entry

### CODE REVIEW

**Comment by MarcoGorelli** [comment]
Thanks @jeandersonbc 

However, before this PR can be accepted, you need to write tests / make sure you don't break existing ones - see [contributing to the code base](https://pandas.pydata.org/pandas-docs/stable/development/contributing.html#contributing-to-the-code-base):

> pandas is serious about testing and strongly encourages contributors to embrace test-driven development (TDD). This development process “relies on the repetition of a very short development cycle: first the developer writes an (initially failing) automated test case that defines a desired improvement or new function, then produces the minimum amount of code to pass that test.” So, before actually writing any code, you should write your tests. Often the test can be taken from the original GitHub issue. However, it is always worth considering additional use cases and writing corresponding tests.
> 
> Adding tests is one of the most common requests after code is pushed to pandas. Therefore, it is worth getting in the habit of writing tests ahead of time so this is never an issue.

**Comment by jeandersonbc** [comment]
Thanks for the feedback @MarcoGorelli! I'm wondering whether the example provided in the related issue (first comment in https://github.com/pandas-dev/pandas/issues/32073) would be sufficient to be added as a test in the `TestDataFramePlots` (test_frame.py) class. I started the discussion on tests on the issue yesterday (although here might be more appropriated).

File: pandas/plotting/_matplotlib/core.py [hunk 1]
```diff
@@ -413,7 +413,13 @@ def _compute_plot_data(self):
         # np.ndarray before plot.
         numeric_data = numeric_data.copy()
         for col in numeric_data:
-            numeric_data[col] = np.asarray(numeric_data[col])
+
+            # GH32073: cast to float if values contain nulled integers
+            values = numeric_data[col]
+            if values.isna().any().all():
+                values = values.astype(float)
+
```
Comment (TomAugspurger) [inline pandas/plotting/_matplotlib/core.py]: It seems like date-like data can be included here, and we don't want to convert those to floats.

I think this should be restricted to

```python
if is_integer_dtype(values.dtype):
    values = values.to_numpy(dtype="float", na_value=np.nan)
```

**Review by TomAugspurger — COMMENTED** [review]
Thanks for working on this. We'll need a test and a release note in 1.1.0.rst.

**Comment by jeandersonbc** [comment]
well, I'm going to close this pull request and work in a single branch as this was not the first time that I messed up after updating my branch

**Pull request closed without merging**